timely_beliefs/beliefs/utils.py

```diff
@@ -313,20 +312,33 @@ def resample_event_start(
 def load_time_series(
     event_value_series: pd.Series,
     sensor: Sensor,
-    source: BeliefSource,
-    belief_horizon: timedelta,
+    source: Union[BeliefSource, pd.Series],
+    belief_horizon: Union[timedelta, pd.Series],
     cumulative_probability: float = 0.5,
 ) -> List["classes.TimedBelief"]:
     """Turn series entries into TimedBelief objects."""
     beliefs = []
-    for time, value in event_value_series.items():
+    if isinstance(belief_horizon, timedelta):
+        belief_horizon_series = pd.Series(belief_horizon)
+    else:
+        belief_horizon_series = belief_horizon
+    if isinstance(source, BeliefSource):
+        source_series = pd.Series(BeliefSource)
+    else:
+        source_series = source
```

Comment:
I guess we should check if `source_series` end up with the same number of elements as `event_value_series`.